modules/transports/optional/fix/src/test/java/org/apache/synapse/transport/fix/FIXTransportListenerTest.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.description.AxisService;
+import org.apache.axis2.description.Parameter;
+import org.apache.axis2.description.TransportInDescription;
+import org.apache.axis2.description.TransportOutDescription;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.axis2.transport.testkit.axis2.TransportDescriptionFactory;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.ArgumentMatchers.anyString;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("javax.management.*")
+@PrepareForTest(FIXTransportListener.class)
+public class FIXTransportListenerTest extends TestCase {
+    @Mock(name = "sessionFactory")
+    FIXSessionFactory fixSessionFactory;
+
+    @InjectMocks
+    FIXTransportListener spy;
+
+    private static Log logger = LogFactory.getLog(FIXTransportListenerTest.class.getName());
+    @Test
+    public void testFIXTransportListenerInit() throws Exception {
+        AxisService axisService = new AxisService("testFIXService");
+        axisService
+                .addParameter(new Parameter(FIXConstants.FIX_ACCEPTOR_CONFIG_URL_PARAM, "/sample/path/Mock.cfg"));
+        axisService.addParameter(
+                new Parameter(FIXConstants.FIX_INITIATOR_CONFIG_URL_PARAM, "/sample/path/Mock2.cfg"));
+
+        ConfigurationContext cfgCtx = new ConfigurationContext(new AxisConfiguration());
+        TransportDescriptionFactory transportDescriptionFactory = new TransportDescriptionFactory() {
+            @Override
+            public TransportOutDescription createTransportOutDescription() throws Exception {
+                TransportOutDescription trpOutDesc = new TransportOutDescription("fix");
+                trpOutDesc.setSender(new FIXTransportSender());
+                return trpOutDesc;
+            }
+
+            @Override
+            public TransportInDescription createTransportInDescription() throws Exception {
+                TransportInDescription trpInDesc = new TransportInDescription("fix");
+                trpInDesc.setReceiver(new FIXTransportListener());
+                return trpInDesc;
+            }
+        };
+        TransportInDescription transportInDescription = null;
+        try {
+            transportInDescription = transportDescriptionFactory.createTransportInDescription();
+        } catch (Exception e) {
+            logger.error(e);
+            Assert.fail("Error occurred while creating transport in description");
```

Comment:
Why do we need Assert.fail here?

Reply:
If the transport-in-description cannot be created, the test case should fail. That's why I added that.

Reply:
Listener init success scenario missing in the test case. Is there a possibility to write failure init as a separate test case and verify with `expectedExceptions = ` rather do assertion in the catch clause.

Reply:
There is no obvious way to cover the init scenario in a unit test. This test just checks whether the flow of listener init is successful. This assert fail is there to purposefully fail the test case in case transport-in-description cannot be created.